pysam/libcbcf.pyi

```diff
@@ -0,0 +1,457 @@
+import sys
+from typing import (
+    Optional,
+    Union,
+    Any,
+    Sequence,
+    Tuple,
+    Iterator,
+    List,
+    Iterable,
+    Dict,
+    overload,
+    TypeVar,
+    Mapping,
+)
+
+if sys.version_info < (3, 8):
+    from typing_extensions import Literal
+else:
+    from typing import Literal
+
+from pysam.libchtslib import HTSFile
+
+_D = TypeVar("_D")
+
+class VariantHeaderRecord:
+    @property
+    def header(self) -> VariantHeader: ...
+    @property
+    def type(self) -> Optional[str]: ...
+    @property
+    def key(self) -> Optional[str]: ...
+    @property
+    def value(self) -> Optional[str]: ...
+    @property
+    def attrs(self) -> Sequence[Tuple[str, str]]: ...
+    def __len__(self) -> int: ...
+    def __bool__(self) -> bool: ...
+    def __getitem__(self, key: str) -> str: ...
+    def __iter__(self) -> str: ...
+    def get(self, key: str, default: _D = ...) -> Union[_D, str]: ...
+    def __contains__(self, key: str) -> bool: ...
+    def iterkeys(self) -> Iterator[str]: ...
+    def itervalues(self) -> Iterator[Optional[str]]: ...
+    def iteritems(self) -> Iterator[Tuple[str, Optional[str]]]: ...
+    def keys(self) -> List[str]: ...
+    def items(self) -> List[Tuple[str, Optional[str]]]: ...
+    def values(self) -> List[Optional[str]]: ...
+    def update(self, items: Union[Iterable, Dict] = ..., **kwargs) -> None: ...
+    def pop(self, key: str, default: str = ...) -> str: ...
+    # def remove(self) -> Any: ...  # crashes
+
+class VariantHeaderRecords:
```

Comment:
Sequence[VariantHeaderRecord]?  May need to add ABC base class to get mixins (__contains__, __reversed__, index, count).

Reply:
Agreed, also to remaining comments. I’ll need to have a look at collections.abc to see what can be used.

Reply:
I didn’t want to touch the implementation, so I haven’t actually changed this to derive from `Sequence` (the mixins are missing as you noted). That can be done at a later stage.